File: backend/buildeR.DAL/Entities/BuildPlugin.cs
using buildeR.DAL.Entities.Common;

using System.Collections.Generic;

namespace buildeR.DAL.Entities
{
    public class BuildPlugin : Entity
    {
        public string PluginName { get; set; }
        public string Command { get; set; }
        public string DockerImageName { get; set; }
        public string DockerRegistryName { get; set; }
        public virtual IEnumerable<PluginCommand> PluginCommands { get; set; }
    }
}

File: frontend/src/app/shared/models/build-plugin.ts
import { PluginCommand } from './plugin-command';

export interface BuildPlugin {
  id: number;
  pluginName: string;
  command: string;
  dockerImageName: string;
  version: string;
  dockerRegistryName: string;
  isCollapsed: boolean;
  pluginCommands: PluginCommand[];
  newCommand: string;
}
